scripts/linter/collect-docs.js

```diff
@@ -0,0 +1,47 @@
+const fsPromises = require("fs").promises;
+const path = require("path");
+
+/**
+ * Walks the content directory tree (`start`) yielding Markdown or `meta.yaml` files with `recipe` frontmatter.
+ */
+async function* walkContent(start = "content") {
+    const files = await fsPromises.readdir(start, { withFileTypes: true });
+
+    const hasMeta = files.map(f => f.name).includes("meta.yaml");
+    if (hasMeta) {
+        yield path.join(start, "meta.yaml");
+    } else {
+        for (const f of files) {
+            if (f.name.endsWith(".md")) {
+                yield path.join(start, f.name);
+            } else if (f.isDirectory()) {
+                yield* walkContent(path.join(start, f.name));
+            }
+        }
+    }
+}
+
+/**
+ * Checks if there's a recipe name in `filePath`.
+ */
+async function isRootDoc(filePath) {
+    const file = await fsPromises.readFile(filePath);
+    return /^recipe: [\w-]+$/gm.test(file);
```

Comment:
By definition, the correct answer (at the moment) to "is this file the place to start building a thing" is "this is a Markdown file containing front matter, that contains a `recipe` property".

Because that's how build-json decides what it's going to try to build. So ideally the linter should use exactly the same definition.

But I don't think this is a blocker for the linter at this point anyway.

>  The naive approach read each Markdown file, walked its tree, and parsed its YAML two times: once to identify the Markdown file as having a recipe and once to run checks on it. Avoiding the repetition would've introduced a lot of complexity that the regex sidestepped.

Yes, build-json is super-inefficient and eventually that will become a problem. I wonder if vfiles could help us here?